src/vs/base/common/uri.ts

```diff
@@ -73,8 +74,17 @@ function _referenceResolution(scheme: string, path: string): string {
 
 const _empty = '';
 const _slash = '/';
+
 const _regexp = /^(([^:/?#]+?):)?(\/\/([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/;
 
+const enum MatchIndex {
```

Comment:
Question: why don't you use regex group?

Reply:
must run in IE which doesn't support this